### ISSUE TITLE
Are past_due subscriptions active or not ? 

### ISSUE THREAD

**Issue by clemoun** [pay-rails/pay#529]
One thing that is quite confusing regarding `past_due` subscriptions is that :
 - They will be returned in the `:active` scope
 - But they will give a false with the `active?` method

My proposition would be that the `:active` scope should be : `scope :active, -> { where(ends_at: nil, status: ["trialing", "active"]).or(on_grace_period).or(on_trial) }`

and we could have an `:on_going` scope with  `scope :active, -> { where(ends_at: nil, ).or(on_grace_period).or(on_trial) }`

**Comment by excid3** [comment]
Thanks for pointing this out @clemoun. Those definitely should be consistent.

I think it makes sense if `past_due` subscriptions are not considered `active`. That way you can warn users they need to complete their payment in order to continue using the product.

* On Trial
* Active - on trial, currently active, or marked to cancel in the future
* Past Due

We could introduce an `active_or_past_due` scope if anyone wants to preserve that functionality where it includes both.

Does that make sense to you?

**Comment by clemoun** [comment]
That would indeed be clearer this way !

One question though (sorry, noob here). If you update the `:active` scope on paygem and push it to production, when will this change affect my own app with paygem in the Gemfile ?

**Comment by excid3** [comment]
It would, so we would need to make a version bump that's more than a bugfix if we change that since it would be a somewhat breaking change.

**Comment by clemoun** [comment]
Hello again !
I realized that subscriptions with `incomplete` status are also included in the `active` scope. This also does not seem right as it will grant access to the product for every customers that fail the payment process...

**Comment by clemoun** [comment]
If I can help in any way to propose a code update to fix all this, let me know. I'm not sure how to do this properly

**Comment by excid3** [comment]
Correct, no statuses were included in the scope. I have pushed 5e4c060 for this.

**Comment by garethfr** [comment]
This logic still seems to be incorrect. I have the following subscription:

`#<Pay::Subscription id: 21, customer_id: 29, name: "default", processor_id: "xxx", processor_plan: "xxx", quantity: 1, status: "trialing", trial_ends_at: "2022-02-13 21:36:39.081256000 +0000", ends_at: nil, application_fee_percent: nil, metadata: {}, data: {}, created_at: "2022-02-14 21:34:56.102039000 +0000", updated_at: "2022-02-14 21:36:39.116513000 +0000", prorate: true>`

`:trial_ends_at` is in the past, but `:ends_at` is still nil, so it is still being considered as `:active`

This subscription was created with `Subscriptions.new(current_user).create_trial`

**Comment by excid3** [comment]
If a trial has ended and there is no end date, that is correct. The status is mostly informational. Your status should be "active" in this case, but Braintree and other payment processors don't have statuses that exactly match Stripe or even webhooks to update those.
